src/kbmod/search/raw_image.cpp

```diff
@@ -238,239 +187,123 @@ void RawImage::convolve_cpu(const PSF& psf) {
             for (int j = -psf_rad; j <= psf_rad; j++) {
                 for (int i = -psf_rad; i <= psf_rad; i++) {
                     if ((x + i >= 0) && (x + i < width) && (y + j >= 0) && (y + j < height)) {
-                        float current_pixel = pixels[(y + j) * width + (x + i)];
+                        float current_pixel = image(y + j, x + i);
+                        // note that convention for index access is flipped for PSF
```

Comment:
My confusion is that we don't have a PSF class (since PSF would be the function that is fitted, so a bunch of coefficients). We have a PSF class that holds a single realization of a PSF on some sized-grid, given that I'm not sure how it fits in with Eigen I left it out for now. Same problems as with RawImage, it's kind of like a wrapper around a thing the Matrix already is by itself. You want that to be a part of this PR?